src/skmultiflow/data/temporal_data_stream.py

```diff
@@ -0,0 +1,153 @@
+import pandas as pd
+import numpy as np
+
+import warnings
+
+from skmultiflow.data.data_stream import DataStream
+from skmultiflow.utils import add_delay_to_timestamps
+
+class TemporalDataStream(DataStream):
+    """ Creates a temporal stream from a data source.
+
+    TemporalDataStream takes the whole data set containing the `X` (features), `time` (timestamps) and `Y` (targets).
+
+    Parameters
+    ----------
+    data: np.ndarray or pd.DataFrame
+        The features' columns and targets' columns or the feature columns
+        only if they are passed separately.
+    time: np.ndarray(dtype=datetime64) or pd.Series (Default=None)
+        The timestamp column of each instance. If its a np.ndarray, it will
+        be converted into a pandas datetime dataframe. 
+    sample_weight: np.ndarray or pd.Series, optional (Default=None)
+        Sample weights.
+    sample_delay: np.ndarray(np.datetime64), pd.DataFrame, np.timedelta64 or int, optional (Default=np.timedelta64(0,"D"))
+        Samples delay in np.timedelta64 (the dateoffset difference between the event time
+        and when the label is available), np.ndarray(np.datetime64) with the timestamp that the sample
+        will be available or int with the delay in number of samples (similar to MOA, a sample is just processed
+        after a number of sample_delay samples).
+    y: np.ndarray or pd.DataFrame, optional (Default=None)
+        The targets' columns.
+    target_idx: int, optional (default=-1)
+        The column index from which the targets start.
+
+    n_targets: int, optional (default=1)
+        The number of targets.
+
+    cat_features: list, optional (default=None)
+        A list of indices corresponding to the location of categorical features.
+
+    name: str, optional (default=None)
+        A string to id the data.
+
+    ordered: bool, optional (default=True)
```

Comment:
I am not sure about this, it would be much easier for the user but always would take more time considering the default parameters. Do you think that setting it to `False` given this condition is ok?